scripts/install/opt/seagate/cortx/motr/libexec/motr_data_recovery.sh

```diff
@@ -817,6 +819,28 @@ run_becktool() {
         # restart the execution of command if exit code is ESEGV error
         while [[ $cmd_exit_status == $ESEGV ]];
         do
+            #Following is the code to limit the number of core-m0beck and m0trace files, generated due to m0beck crash( receving SEGV signal ), to 2 each.
+            core_m0beck_file_count=$(cd /var/log/crash; find . -type f -exec stat  -c "%n %y" {} \;| sort -n | grep core-m0beck | awk '{if($2>"'"${SCRIPT_START_DATE}"'") print $2" "$3; else if($2=="'"${SCRIPT_START_DATE}"'" && $3>="'"${SCRIPT_START_TIME}"'") print $2" "$3; else echo }' | wc -l)
+            echo "File count value $core_m0beck_file_count"
+
+            if [[ $core_m0beck_file_count -gt 2 ]]; then
+                echo "Deleting core m0beck extra file $core_m0beck_file_count"
+                core_m0beck_last_file_name=$(cd /var/log/crash; ls -ltr | grep core-m0beck | awk '{print $9}' | tail -n 1)
+                echo "$core_m0beck_last_file_name"
+                rm "/var/log/crash/$core_m0beck_last_file_name"
+            fi
```

Comment:
Improvements:
1. The logic to get "core_m0beck_file_count" is duplicated multiple times, at least 6 times.
    Please make a function to do that.
2. Define a variable for "var/log/crash"
3. Define a variable for the number (i.e. 2, or 3) for crash files and trace files. Don't use 
    literal number in multiple places.
4. date +"%s" will return the seconds since epoch
    stat -c "%n %Y" /some/file will return "time of last modification, seconds since Epoch"
    It is easier to compare two numbers.
5. Maybe we can "beck" can create a temp file at the beginning and then use "find  -cnewer the_temp_file_beck_created" to find the crash or trace file.

Reply:
Hi @huanghua78 
I have addressed your comment
Commits 919408d and 67841ad are related to the improvements suggested by you.
Point 4 kind of invalidates 5, as it reduces the number of comparisons to be made. :)